### PR TITLE
Unstub the call to create filter api

### PR DESCRIPTION
**Pull request by mattrout92**
### What

The real filter api has been hooked into the filter-dataset-controller so you can successfully add/remove dimension options to your filter job

### How to review

Start the following services:

- dp-frontend-router
- dp-frontend-dataset-controller
- dp-frontend-filter-dataset-controller (feature/unstubination)
- dp-frontend-router (feature/unstubination)
- zebedee
- babbage
- sixteens
- dp-hierarchy-api
- dp-codelist-api
- dp-filter-api

Navigate to a dataset landing page. For example:

http://localhost:20000/employmentandlabourmarket/peopleinwork/workplacedisputesandworkingconditions/datasets/labourdisputesbysectorlabd02

Click filter and download. Verify that a filter job has been created.

Click on the time dimension and verify that you can add/remove (including all) dimension options for both the range selector and the list selector. Verify that when you return to the filter options, the number of selected options are visible next to the time dimension.

Please be aware that there is a bug on the filter-api where too many connections are being opened and not closed to the database. If you get an internal error you should restart the filter api until this is fixed.

### Who can review

Anyone

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -5,16 +5,18 @@ import "github.com/ian-kent/gofigure"
 var cfg Config
 
 type Config struct {
-	BindAddr    string `env:"BIND_ADDR"`
-	ZebedeeURL  string `env:"ZEBEDEE_URL"`
-	RendererURL string `env:"RENDERER_URL"`
+	BindAddr     string `env:"BIND_ADDR"`
+	ZebedeeURL   string `env:"ZEBEDEE_URL"`
+	RendererURL  string `env:"RENDERER_URL"`
+	FilterAPIURL string `env:"FILTER_API_URL"`
```
Comment (nshumoogum) [inline config/config.go]: No flags (and hence flag descriptions)?

File: filter/filter.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+// New creates a new instance of Client with a given filter api url
+func New(filterAPIURL string) *Client {
+	return &Client{
+		cli: &http.Client{Timeout: 5 * time.Second},
+		url: filterAPIURL,
+	}
+}
+
+//FilterJob ...
```
Comment (nshumoogum) [inline filter/filter.go]: go documentation?

File: filter/filter.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+type FilterJob struct {
+	DatasetFilterID string `json:"dataset_filter_id"`
+	FilterJobID     string `json:"filter_job_id"`
+	State           string `json:"state"`
+}
+
+// CreateJob creates a filter job and returns the associated filterJobID
+func (c *Client) CreateJob(datasetFilterID string) (string, error) {
+	fj := FilterJob{DatasetFilterID: "384900328", State: "created"} // DatasetFilterId comes from dataset api
```
Comment (nshumoogum) [inline filter/filter.go]: fj? filterJob would be better

File: handlers/handlers.go [hunk 4]
```diff
@@ -32,14 +31,21 @@ func init() {
...
-	// code has been written
-	filterID := rand.Intn(100000000)
-	fid := strconv.Itoa(filterID)
-
-	log.Trace("created filter id", log.Data{"filter_id": fid})
-	http.Redirect(w, req, "/filters/"+fid+"/dimensions", 301)
+func CreateFilterID(fil *filter.Client) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+
+		fid, err := fil.CreateJob("ds83jks0-23euufr89-8i3") //use a real dataset filter id
```
Comment (nshumoogum) [inline handlers/handlers.go]: Leave TODO comment, instead a general comment in code base. It'll be easier to search for when cleaning up service.

Comment (nshumoogum) [inline handlers/handlers.go]: Also shouldn't fil just be client?

File: config/config.go (hunk 1 above)
Comment (mattrout92) [inline config/config.go]: yeah, they don't use flags at all for configuration

File: filter/filter.go (hunk 3 above)
Comment (mattrout92) [inline filter/filter.go]: https://github.com/golang/go/wiki/CodeReviewComments#variable-names

Comment (nshumoogum) [inline filter/filter.go]: Never mind then 👍

**Pull request merged**